### PR TITLE
Create vmtest.log lazily

### PR DESCRIPTION
**Pull request by danielocfb**
It is super annoying to have an empty .vmtest.log dropped anywhere the program has been run. Provide some logic that makes it so that the log file is created lazily and only when data is actually written to it.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -43,6 +44,57 @@ struct Args {
...
+
+impl DeferredLog {
+    fn file(&mut self) -> io::Result<&File> {
+        self.file.get_or_try_init_(|| {
+            fs::OpenOptions::new()
+                .create(true)
+                .append(true)
+                .open(".vmtest.log")
+        })
+    }
```
Comment (danobi) [inline src/main.rs:83]: What happens if the file fails to create? Does `log` print a warning and then continue along? 

Maybe it's better/simpler to `unwrap()` the file creation. At least that way you wouldn't need to have the try init dance

Comment (danielocfb) [reply src/main.rs:83]: It would be bubbled up through the `write` / `flush` call. We can also unwrap. I don't have a strong preference.

Comment (danobi) [reply src/main.rs:83]: Right, but in our code which just does `info!(...)` we are not checking for errors or anything. If that happens to just panic, we may as well panic during file creation here.

Comment (d-e-s-o) [reply src/main.rs:83]: I checked and it does not panic but just swallow any errors.

Comment (danobi) [reply src/main.rs:83]: Thanks for checking. In that case we'd probably want to panic - silently dropping logs is not very good

Comment (d-e-s-o) [reply src/main.rs]: Sure thing. Changed.

**Review by danobi — APPROVED** [review]

**Pull request merged**